arch/src/x86_64/mod.rs

```diff
@@ -768,6 +772,20 @@ pub fn configure_vcpu(
     CpuidPatch::set_cpuid_reg(&mut cpuid, 0xb, None, CpuidReg::EDX, u32::from(id));
     CpuidPatch::set_cpuid_reg(&mut cpuid, 0x1f, None, CpuidReg::EDX, u32::from(id));
 
+    if !kvm_hyperv {
```

Comment:
Don't you want to also check whether "Invariant TSC" feature is available?

QEMU patch did it:
```
...
        /* TSC clock must be stable and known for this feature. */
        && ((env->features[FEAT_8000_0007_EDX] & CPUID_APM_INVTSC)
...
```

See https://github.com/qemu/qemu/commit/9954a1582e18b03ddb66f6c892dccf2c3508f4b2

Reply:
Although I expect anybody using this should check there is not much harm in doing the extra check.